Makefile

```diff
@@ -129,12 +159,24 @@ $(LOCALBIN):
 ## Tool Binaries
 KUSTOMIZE ?= $(LOCALBIN)/kustomize
 CONTROLLER_GEN ?= $(LOCALBIN)/controller-gen
+KIND ?= $(LOCALBIN)/kind
+GINKGO ?= $(LOCALBIN)/ginkgo
 ENVTEST ?= $(LOCALBIN)/setup-envtest
 
 ## Tool Versions
 KUSTOMIZE_VERSION ?= v4.5.7
 CONTROLLER_TOOLS_VERSION ?= v0.10.0
 
+.PHONY: kind
+kind: $(KIND) ## Download kind locally if necessary.
+$(KIND): $(LOCALBIN)
+	test -s $(LOCALBIN)/kind || GOBIN=$(LOCALBIN) go install sigs.k8s.io/kind@v0.15.0
+
+.PHONY: ginkgo
+ginkgo: $(GINKGO) ## Download ginkgo locally if necessary.
+$(GINKGO): $(LOCALBIN)
+	test -s $(LOCALBIN)/ginkgo || GOBIN=$(LOCALBIN) go install github.com/onsi/ginkgo/v2/ginkgo@v2.1.4
```

Comment:
The `test -s $(LOCALBIN)/<binName>` tests are sufficient for CI because they'll always fail and we'll always get the correct version. 

But locally, these can be annoying when we bump the version of one of these and you still have an old version from an earlier commit. If there's some diff in behavior between your local version and the one specified here, you'll be scratching your head trying to figure out why.

There are two solutions I've seen for this:
1. Rukpak has a separate [`tools` go module](https://github.com/operator-framework/rukpak/tree/main/hack/tools) with a separate go.mod that lists all the tool versions with [make targets](https://github.com/operator-framework/rukpak/blob/c527e657b4dabfd228068fbe458b72acb013bf00/Makefile#L223-L258) that always `go build`
2. A [somewhat curated script](https://github.com/operator-framework/helm-operator-plugins/blob/main/tools/scripts/fetch) that knows how to check the presence and version of the existing binary to see if a new one is needed [called like this](https://github.com/operator-framework/helm-operator-plugins/blob/dd32bac92534db6a00fe30252cd944c361639ec5/Makefile#L81). It's _mostly_ set and forget.

Option 1 is good if all of our tools are Go-based
Option 2 is probably slightly faster when actually using the tools, but requires some tweaks every now and then when a tool changes its release artifacts.

Reply:
This could be a follow-up for sure.

Reply:
I did notice number 1 there in rukpak's makefile. I initially was working with that but it ended up turning into a bit of a rabbit-hole with the amount of changes needed, since it appeared as though it would require re-working every built-in make target we got from kubebuilder, and then at that point it felt like I was going to end up with just a copy-paste of rukpak's makefile which would require re-testing all of the targets. We definitely need to take care of it at some point, if you think it's a good thing to take care of now I could give it another go. It just felt like it was getting a bit out-of-scope to me.

Reply:
I've created issue #85 to track this.